src/Auth0.Core/Exceptions/RateLimitApiException.cs

```diff
@@ -46,15 +52,17 @@ public RateLimitApiException(string message, Exception innerException)
         /// Initializes a new instance of the <see cref="RateLimitApiException"/> class with a specified <paramref name="rateLimit"/>.
         /// </summary>
         /// <param name="rateLimit"><see cref="Exceptions.RateLimit"/> received on the API call that failed.</param>
-        public RateLimitApiException(RateLimit rateLimit)
+        /// <param name="apiError"><see cref="Exceptions.ApiError"/> received on the API call that failed.</param>
+        public RateLimitApiException(RateLimit rateLimit, ApiError apiError)
```

Comment:
In order to stay backwards compatible, this constructor should take an optional ApiError.
Even though a user is not expected to be instantiating this class themselves, I think it's a good idea to not break existing APIs when not neccessary.

Reply:
Good point. We cannot assume that nobody else is using that constructor.
I'll make the changes in a second.
re integration tests: Does it should run against any auth0 tenants? Or does it require a tenant with specific config? I couldn't find the instructions of how to set that up. Thanks

Reply:
>re integration tests: Does it should run against any auth0 tenants? Or does it require a tenant with specific config? I couldn't find the instructions of how to set that up. Thanks

Ye it needs some enterprise setup that not everyone is able to set up.